tests/Neo.UnitTests/SmartContract/Native/UT_CryptoLib.cs

```diff
@@ -427,5 +436,474 @@ public void TestKeccak256_BlankString()
+
+            // Retrieve executing transaction hash.
+            vrf.EmitSysCall(ApplicationEngine.System_Runtime_GetScriptContainer); // push the script container (executing transaction, actually).
+            vrf.Emit(OpCode.PUSH0, OpCode.PICKITEM); // pick 0-th transaction item (the transaction hash).
+
+            // Concatenate network magic and transaction hash.
+            vrf.Emit(OpCode.CAT); // this instruction will convert network magic to bytes using BigInteger rules of conversion.
+
+            // Continue construction of 'verifyWithECDsa' call.
+            vrf.Emit(OpCode.PUSH4, OpCode.PACK); // pack arguments for 'verifyWithECDsa' call.
+            EmitAppCallNoArgs(vrf, CryptoLib.CryptoLib.Hash, "verifyWithECDsa", CallFlags.None); // emit the call to 'verifyWithECDsa' itself.
+
+            // Account is a hash of verification script.
+            var vrfScript = vrf.ToArray();
+            var acc = vrfScript.ToScriptHash();
+
+            var tx = new Transaction
+            {
+                Attributes = [],
+                NetworkFee = 1_0000_0000,
+                Nonce = (uint)Environment.TickCount,
+                Script = new byte[Transaction.MaxTransactionSize / 100],
+                Signers = [new Signer { Account = acc }],
+                SystemFee = 0,
+                ValidUntilBlock = 10,
+                Version = 0,
+                Witnesses = []
+            };
+            var tx_signature = Crypto.Sign(tx.GetSignData(TestBlockchain.TheNeoSystem.Settings.Network), privkey, ECCurve.Secp256k1, Hasher.Keccak256);
+
+            // inv is a builder of witness invocation script corresponding to the public key.
+            using ScriptBuilder inv = new();
+            inv.EmitPush(tx_signature); // push signature.
+
+            tx.Witnesses =
+            [
+                new Witness { InvocationScript = inv.ToArray(), VerificationScript = vrfScript }
+            ];
+
+            tx.VerifyStateIndependent(TestProtocolSettings.Default).Should().Be(VerifyResult.Succeed);
+
+            var snapshot = TestBlockchain.GetTestSnapshot();
+
+            // Create fake balance to pay the fees.
+            ApplicationEngine engine = ApplicationEngine.Create(TriggerType.Application, null, snapshot, settings: TestBlockchain.TheNeoSystem.Settings, gas: long.MaxValue);
+            _ = NativeContract.GAS.Mint(engine, acc, 5_0000_0000, false);
+            snapshot.Commit();
+
+            var txVrfContext = new TransactionVerificationContext();
+            var conflicts = new List<Transaction>();
+            tx.VerifyStateDependent(TestProtocolSettings.Default, snapshot, txVrfContext, conflicts).Should().Be(VerifyResult.Succeed);
+
+            // The resulting witness verification cost is 2154270 GAS.
+            // The resulting witness Invocation script (66 bytes length):
+            // NEO-GO-VM > loadbase64 DEARoaaEjM/3VulrBDUod7eiZgWQS2iXIM0+I24iyJYmffhosZoQjfnnRymF/7+FaBPb9qvQwxLLSVo9ROlrdFdC
```

Comment:
```suggestion
            // NEO-VM > loadbase64 DEARoaaEjM/3VulrBDUod7eiZgWQS2iXIM0+I24iyJYmffhosZoQjfnnRymF/7+FaBPb9qvQwxLLSVo9ROlrdFdC
```

Reply:
The tool that generated these dumps has a specific prompt that is _not_ the one after your edits. If we care a tiny bit about reproducibility of these they shouldn't be touched.